submissions/pashapushak1996/oop-exercise/index.js

```diff
@@ -0,0 +1,57 @@
+class Inhabitant {
+    constructor(name, gender, greeting) {
+        this.species = this.constructor.name.toLowerCase();
+        this.name = name;
+        this.gender = gender;
+        this.greeting = greeting;
+    }
+
+    valuesToString(inhabitantProps) {
+        return inhabitantProps
```

Comment:
Check OOP encapsulation and SOLID open-closed principles.
No need in external variable. A class should be self-sufficient.
Also every class should handle only properties it owns.

Method overloading and extended `super` API to the rescue.